### PR TITLE
Implement __getitem__ inference for classes (using the metaclass)

### LINKED ISSUE #348: __getitem__ inference for classes (using the metaclass)
Ideally, we should be able to infer this:

```
from astroid import extract_node
n = extract_node('''
"""Hello"""
from __future__ import (
    absolute_import,
    division,
    print_function,
    unicode_literals,
)
from six import add_metaclass

class Meta(type):

    def __getitem__(cls, arg):
        return arg

@add_metaclass(Meta)
class Works(object):
    pass

    def __getitem__(self, arg): return arg



x = Works['hello']
x
''')
i = next(n.infer())
print(i)
```

### PR DESCRIPTION
**Pull request by euresti**
Essentially, implement the getitem method in ClassDef which returns the correct value.

Fixes #348

### CODE REVIEW

File: astroid/scoped_nodes.py [hunk 1]
```diff
@@ -1509,6 +1510,29 @@ def _valid_getattr(node):
...
+
+        This is basically looking up the method in the metaclass and calling it.
+        """
+        try:
+            methods = dunder_lookup.lookup(self, '__getitem__')
+        except exceptions.AttributeInferenceError as error:
+            util.reraise(exceptions.InferenceError(**vars(error)))
+
+        method = methods[0]
+        # Rewrap index to Const for this case
```
Comment (PCManticore) [inline astroid/scoped_nodes.py]: This comment does not seem true in this context, since we are not wrapping anything in Const.

**Comment by euresti** [comment]
Hmm that's copy pasted from another method, should I fix the comment there
too?

**Comment by euresti** [comment]
Fixed the comment. What do you think?

**Comment by PCManticore** [comment]
Thank you for this! I merged this manually.

**Pull request closed without merging**